packages/api/src/controllers/user.ts

```diff
@@ -326,23 +355,55 @@ app.post("/", validatePost("user"), async (req, res) => {
 
 app.patch(
   "/:id/suspended",
+  validatePost("suspend-user-payload"),
   authMiddleware({ anyAdmin: true }),
   async (req, res) => {
+    const { suspended, emailTemplate } = req.body as SuspendUserPayload;
     const { id } = req.params;
     const user = await db.user.get(id);
     if (!user) {
-      res.status(404);
-      return res.json({ errors: ["not found"] });
-    }
-    if (req.body.suspended === undefined) {
-      res.status(400);
-      return res.json({ errors: ["suspended field required"] });
+      return res.status(404).json({ errors: ["not found"] });
     }
-    logger.info(
-      `set user ${id} (${user.email}) suspended ${req.body.suspended}`
-    );
+    const { email } = user;
+
+    logger.info(`set user ${id} (${email}) suspended ${suspended}`);
+    await db.user.update(id, { suspended });
+
+    suspendUserStreams(req, id, suspended).catch((err) => {
+      logger.error(
+        `error suspending user streams id=${id} email=${email} err=${err}`
+      );
+    });
 
-    await db.user.update(id, { suspended: !!req.body.suspended });
+    if (suspended && emailTemplate === "copyright") {
+      const {
+        frontendDomain,
+        supportAddr,
+        sendgridTemplateId,
+        sendgridApiKey,
+      } = req.config;
+      try {
+        await sendgridEmail({
+          email,
```

Comment:
Should we CC also infraservice as we are manually doing right now?

Reply:
Ah, right, forgot about that